### PR TITLE
Add system test for insert/fetch of rows w/ nested schema.

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #3118 as a base.~~

Derived from schema / data from @tswast.

Toward #3029.

### CODE REVIEW

**Comment by daspecster** [comment]
@tseaver should be able to merge #3118 and rebase this off master.

Also, there's a lint error in this branch just FYI.

File: system_tests/bigquery.py [hunk 1]
```diff
@@ -673,8 +670,58 @@ def test_create_table_nested_schema(self):
...
+
+        errors = table.insert_data(to_insert)
+        self.assertEqual(len(errors), 0)
+
+        retry = RetryResult(_has_rows, max_tries=8)
+        fetched = retry(self._fetch_single_page)(table)
+        self.assertEqual(len(fetched), len(to_insert))
+
+        for found, expected in zip(sorted(fetched), sorted(to_insert)):
+            self.assertEqual(found[0], expected[0])            # Name
```
Comment (dhermes) [inline system_tests/bigquery.py]: Can you make an assertion on `len(found) == len(expected) == 8`?

File: system_tests/bigquery.py [hunk 2]
```diff
@@ -673,8 +670,58 @@ def test_create_table_nested_schema(self):
...
+            self.assertEqual(found[1], int(expected[1]))       # Age
```
Comment (dhermes) [inline system_tests/bigquery.py]: Do we control `expected`? If yes, why can't we just store an `int`?

File: system_tests/bigquery.py [hunk 3]
```diff
@@ -673,8 +670,58 @@ def test_create_table_nested_schema(self):
...
+                    e_spell['LastUsed'], '%Y-%m-%d %H:%M:%S UTC')
+                e_used = datetime.datetime(*parts[0:6], tzinfo=UTC)
+                self.assertEqual(f_spell['LastUsed'], e_used)
+                self.assertEqual(f_spell['DiscoveredBy'],
+                                 e_spell['DiscoveredBy'])
+                self.assertEqual(f_spell['Properties'], e_spell['Properties'])
+
+                e_icon = base64.standard_b64decode(
+                    e_spell['Icon'].encode('ascii'))
+                self.assertEqual(f_spell['Icon'], e_icon)
```
Comment (dhermes) [inline system_tests/bigquery.py]: Why does this have to be so involved? Why can't we just say `found[4] == expected[4]`?

File: system_tests/bigquery_test_data.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
+[
```
Comment (dhermes) [inline system_tests/bigquery_test_data.json]: Where does this data come from?

File: system_tests/bigquery.py (hunk 1 above)
Comment (tseaver) [inline system_tests/bigquery.py]: I asserted that the lengths were equal above (line 694):  why would I assert a specific length, when that is driven by the data file, and might change if @tswast provides updates?

File: system_tests/bigquery.py (hunk 2 above)
Comment (tseaver) [inline system_tests/bigquery.py]: `expected` is read from disk (data provided by @tswast).  `found` is generated from the resource from the server, using the schema to convert the JSON repr into native Python types.

File: system_tests/bigquery.py (hunk 3 above)
Comment (tseaver) [inline system_tests/bigquery.py]: `found[4]` is the *actual* bytes value, already converted from the JSON-mandated base64-ified text.

File: system_tests/bigquery_test_data.json (hunk 4 above)
Comment (tseaver) [inline system_tests/bigquery_test_data.json]: @tswast's examples, derived from the bigquery doc snippets (not ours, the ones used on cloud.google.com).

**Comment by tseaver** [comment]
@dhermes I think I answered your questions.

**Comment by tseaver** [comment]
@dhermes, @lukesneeringer, @daspecster Any remaining issues?

**Comment by lukesneeringer** [comment]
None from me. :-)

**Pull request merged**